kafka/channel/pkg/reconciler/controller/kafkachannel.go

```diff
@@ -370,6 +388,52 @@ func (r *Reconciler) reconcile(ctx context.Context, kc *v1alpha1.KafkaChannel) e
 	return nil
 }
 
+func (r *Reconciler) reconcileDispatcher(ctx context.Context, dispatcherNamespace string, kc *v1alpha1.KafkaChannel) (*appsv1.Deployment, error) {
+	d, err := r.deploymentLister.Deployments(dispatcherNamespace).Get(dispatcherDeploymentName)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			args := resources.DispatcherArgs{
+				DispatcherNamespace: dispatcherNamespace,
+				Image:               r.dispatcherImage,
+			}
+			expected := resources.MakeDispatcher(args)
+			d, err := r.KubeClientSet.AppsV1().Deployments(dispatcherNamespace).Create(expected)
+			msg := "Dispatcher Deployment created"
+			if err != nil {
+				msg = fmt.Sprintf("not created, error: %v", err)
+			}
+			r.Recorder.Eventf(kc, corev1.EventTypeNormal, dispatcherDeploymentCreated, "%s", msg)
+			return d, err
+		}
+
+		logging.FromContext(ctx).Error("Unable to get the dispatcher Deployment", zap.Error(err))
+		kc.Status.MarkDispatcherFailed("DispatcherDeploymentGetFailed", "Failed to get dispatcher Deployment")
+		return nil, err
+	}
+	return d, err
+}
+
+func (r *Reconciler) reconcileDispatcherService(ctx context.Context, dispatcherNamespace string, kc *v1alpha1.KafkaChannel) (*corev1.Service, error) {
+	svc, err := r.serviceLister.Services(dispatcherNamespace).Get(dispatcherDeploymentName)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			expected := resources.MakeDispatcherService(dispatcherNamespace)
+			svc, err := r.KubeClientSet.CoreV1().Services(dispatcherNamespace).Create(expected)
+			msg := "Dispatcher Service created"
+			if err != nil {
+				msg = fmt.Sprintf("Dispatcher Service not created, error: %v", err)
+			}
+			r.Recorder.Eventf(kc, corev1.EventTypeNormal, dispatcherServiceCreated, "%s", msg)
+			return svc, err
+		}
+
```

Comment:
I think we might want to also check that we own the service, and complain otherwise

Reply:
I'd rather do this in a different PR. I'll open an issue.